gqcp/include/QCModel/HF/UHF.hpp

```diff
@@ -198,13 +207,19 @@ class UHF {
      *
      *  @return the sigma-spin UHF 1-DM expressed in the underlying scalar basis
      */
-    static OneDM<Scalar> calculateScalarBasis1DM(const TransformationMatrix<double>& C_sigma, const size_t N_sigma) {
+    static SpinResolvedOneDM<Scalar> calculateScalarBasis1DM(const TransformationMatrix<double>& C_a, const TransformationMatrix<double>& C_b, const size_t N_a, const size_t N_b) {
 
-        const auto K_sigma = C_sigma.numberOfOrbitals();
-        const auto D_orthonormal = UHF<Scalar>::calculateOrthonormalBasis1DM(K_sigma, N_sigma);
+        const auto K_a = C_a.numberOfOrbitals();
+        const auto K_b = C_b.numberOfOrbitals();
+        const auto D_orthonormal = UHF<Scalar>::calculateOrthonormalBasis1DM(K_a, K_b, N_a, N_b);
+
+        // Calculate the alpha and beta scalar basis 1 DM
+        OneDM<Scalar> D_AO_a = C_a.conjugate() * D_orthonormal.alpha() * C_a.transpose();
+        OneDM<Scalar> D_AO_b = C_b.conjugate() * D_orthonormal.beta() * C_b.transpose();
+        SpinResolvedOneDM<Scalar> D_AO {D_AO_a, D_AO_b};
```

Comment:
Instead of manually transforming a spin-resolved 1-DM to the AO basis, could you add this API into `SpinResolvedOneDM`?